src/core/src/core_logic/PatchInstaller.py

```diff
@@ -174,39 +174,46 @@ def install_updates(self, maintenance_window, package_manager, simulate=False):
         maintenance_window_exceeded = False
         all_packages, all_package_versions = package_manager.get_all_updates(True)  # cached is fine
         self.telemetry_writer.write_event("All available packages list: " + str(all_packages), Constants.TelemetryEventLevel.Verbose)
-        self.last_still_needed_packages = all_packages
-        self.last_still_needed_package_versions = all_package_versions
+        self.last_still_needed_packages = list(all_packages)
```

Comment:
This change is to copy the existing all_packages list to create new list self.last_still_needed_packages.

As per existing code, both the list i.e. all_packages and self.last_still_needed_packages were pointing to the same list in memory and hence modifying one list was altering the other list.